### ISSUE TITLE
Default exports losing type information

### ISSUE THREAD

**Issue by dwelch2344** [rezonant/phantomdi#3]
Okay, this one might get a bit tricky to explain... 

Basically if a base library (`lib`) that uses phantom re-exports a `default export class` in the "main export" (like `index.ts` in root of project), type information cannot be resolved in a consuming project (`app`) 

Created https://github.com/dwelch2344/phantomdi-public-exports to showcase it. 

# Example Setup

`lib/example.ts`
``` 
export default class ExampleClass {}
```

`lib/index.ts`
```
export { default as ExampleClass } from './example`
```

`app/app.ts`
```
import { ExampleClass } from 'lib'

function doThing(ex: ExampleClass) {
  console.log('do thing', ex)
}


// injector([ provide(ExampleClass) ]).invoke(null, doThing) 
```

Will fail. Interestingly, direct import (ie `import ExampleClass from 'lib/lib/example.js'`) works fine... 

More comprehensive info on the README, but had a few on my lunch so wanted to get this over

**Comment by rezonant** [comment]
Beautiful repro! I will take a look!

**Comment by rezonant** [comment]
OK so the issue stems from:

```javascript
// example-app.js type lib:
[93]: { LΦ: t => require("phantomdi-lib").default }, [14]: { LΦ: t => Number } }
```

However, the correct import is:
```javascript
require("phantomdi-lib").ExampleClass 
```

It looks like typescript-rtti is getting confused about the _original_ export style versus the _final_ export style (since this is a re-export of the default)

**Comment by rezonant** [comment]
Filed https://github.com/typescript-rtti/typescript-rtti/issues/27 upstream

Yes, dealing with cross _package_ testing is difficult, and I don't have a great way to do it yet. But luckily this can be reproduced within a single compilation unit:

```typescript
it('emits correctly for non-default re-export of a default export', async () => {
    let exports = await runSimple({
        trace: true,
        moduleType,
        code: `
            import { A } from "./libf";
            export function f(a : A) { }
            export { A } from "./libf";
        `, 
        modules: {
            './libf.ts': `
                export { default as A } from './a'
            `,
            './a.ts': `
                export default class A {
                    method(hello : lib.A) { return 123; }
                }
            `
        }
    });

    let params = Reflect.getMetadata('rt:p', exports.f);
    expect(params[0].t()).to.equal(exports.A);
});
```

**Comment by rezonant** [comment]
Fantastic work finding this @dwelch2344 :-D Thank you for such a good repro! This should be fixed pretty quick, hopefully tonight.

**Comment by rezonant** [comment]
I believe this is fixed with typescript-rtti@0.4.18, reopen if you see otherwise!
